web/utils.py

```
@@ -33,3 +42,86 @@ def get_or_create_cart(request):
+        # Calculate points for all time periods
+        total_points = calculate_user_total_points(user)
+        weekly_points = calculate_user_weekly_points(user)
+        monthly_points = calculate_user_monthly_points(user)
+        current_streak = calculate_user_streak(user)
+        challenge_count = ChallengeSubmission.objects.filter(user=user).count()
+
+        # Skip users with no points in the relevant period
+        if period == "weekly" and weekly_points <= 0:
+            continue
+        elif period == "monthly" and monthly_points <= 0:
+            continue
```

Ruff (0.8.2)

Combine `if` branches using logical `or` operator

Combine `if` branches

(SIM114)